Library/Homebrew/test/os/dependency_collector_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "dependency_collector"
```

Comment:
Sorry, I may have misread/typed before but this is still the wrong filename: please put it in `test/dependency_collector_spec.rb` and add tests in `test/os/mac/dependency_collector_spec.rb` (these files both already exist)

Reply:
:confused: I'm confused as to what goes where. Coincidentally, tests have to be updated to account for the "conditional nature" of `bzip2` and `zip` dependencies. Shall I make tests for macOS unconditional?

#### Unconditional version that will work on macOS only:
```ruby
    it "creates a resource dependency from a '.bz2' URL" do
      resource = Resource.new
      resource.url("http://example.com/foo.tar.bz2")
      expect(subject.add(resource)).to be nil
    end
```
#### Conditional version that (I think) will work on both macOS and Linux
though, I don't know whether `if which("bzip2")` works in tests.
```ruby
    it "creates a resource dependency from a '.bz2' URL" do
      resource = Resource.new
      resource.url("http://example.com/foo.tar.bz2")
      if which("bzip2")
          expect(subject.add(resource)).to be nil
      else
          expect(subject.add(resource)).to eq(Dependency.new("bzip2", [:build]))
      end
    end
```